### ISSUE TITLE
[Installation]: CRSF Error with Reverse Proxy

### ISSUE THREAD

**Issue by dbanshee** [tubearchivist/tubearchivist#441]
### Installation instructions

- [X] I have read and understand the [installation instructions](https://github.com/tubearchivist/tubearchivist#installing-and-updating).
- [X] My issue is not described in the [common errors](https://github.com/tubearchivist/tubearchivist#common-errors) section.

### Operating System

Linux

### Your installation or setup question

After set Apache as Reverse Proxy SSL y tubearchivist fails on login with CRSF verification failed.

Acording to doc

```
Set the environment variable TA_HOST to match with the system running Tube Archivist. This can be a domain like example.com, a subdomain like ta.example.com or an IP address like 192.168.1.20, add without the protocol and without the port. You can add multiple hostnames separated with a space. Any wrong configurations here will result in a Bad Request (400) response.
```

But revising the commit 
[https://github.com/tubearchivist/tubearchivist/commit/8a1ae1ef8331407bc3e16a863b002998fc2d861d](url)

```
CSRF_TRUSTED_ORIGINS = [i.strip() for i in environ.get("TA_HOST").split()]

CSRF_TRUSTED_ORIGINS = []
for host in ALLOWED_HOSTS:
    if host.startswith("http://") or host.startswith("https://"):
        CSRF_TRUSTED_ORIGINS.append(host)
    else:
        CSRF_TRUSTED_ORIGINS.append(f"http://{host}")
```

Only if I set TA_HOST as `https://mydomain` (with protocol) it will be added to TRUSTED_ORIGINS

After adding this protocol prefix, I can access sucessfully from my Apache Reversed Proxy.

### Relevant log output

```shell
not available
```

**Comment by bbilly1** [comment]
Sorry to let this lying around. You identified the code in question, but to clarify what it does, it will append a fallback protocol of *http://* to the CSRF_TRUSTED_ORIGINS if the TA_HOST doesn't have a environment variable specified. 

But either way, you are right, this isn't handled well. TA_HOST needs without the protocol, but CSRF_TRUSTED_ORIGINS needs the protocol. I was trying to avoid having to specify two separate variables.

So I'd suggest: 
If you have https, then specify the protocol in the TA_HOST environment variable. Then the parser will strip the protocol before adding it to ALLOWED_HOSTS but keep it for CSRF_TRUSTED_ORIGINS. Else if you use http, don't specify the protocol in TA_HOST, and the parser adds it for CSRF_TRUSTED_ORIGINS. Then update the documentation to clarify. 

This way that doesn't become a breaking change but we can still do both with a single environment variable. Does that make sense?

**Comment by lamusmaser** [comment]
Could we add both `http` and `https` to the list? That would allow for either to come through and then non-specified protocols would work as expected.

**Comment by bbilly1** [comment]
OK, so as an example this would mean:

TA_HOST `tubearchivist.local` *or* `http://tubearchivist.local` will result in

ALLOWED_HOSTS: `tubearchivist.local`
CSRF_TRUSTED_ORIGINS: `http://tubearchivist.local`

And 

TA_HOST `https://tubearchivist.local` will result in

ALLOWED_HOSTS: `tubearchivist.local`
CSRF_TRUSTED_ORIGINS: `https://tubearchivist.local`

So that's not a breaking change, and you wouldn't have to but can specify the protocol for http, but https you'll have to add the protocol.

**Comment by lamusmaser** [comment]
Basically, you'll want to have the default be `http` and you'll need to specify if the origin is coming from `https`? I was suggesting the following:

### Scenario One - No Protocol
ALLOWED_HOSTS: `tubearchivist.local`
CSRF_TRUSTED_ORIGINS: `[http://tubearchivist.local, https://tubearchivist.local]`

### Scenario Two - HTTP Specified
ALLOWED_HOSTS: `http://tubearchivist.local`
CSRF_TRUSTED_ORIGINS: `http://tubearchivist.local`

### Scenario Two - HTTPS Specified
ALLOWED_HOSTS: `https://tubearchivist.local`
CSRF_TRUSTED_ORIGINS: `https://tubearchivist.local`

Since it is doing an append action, this should be doable, correct?

**Comment by bbilly1** [comment]
This should now be fixed in v0.3.5, please update.
